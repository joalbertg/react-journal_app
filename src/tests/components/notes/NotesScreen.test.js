import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { act } from '@testing-library/react';

import NoteScreen from '~components/notes';
import {
  activeNote,
  startDeleting
} from '~actions';

jest.mock('~actions', () => ({
  activeNote: jest.fn(),
  startDeleting: jest.fn()
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const note = {
  id: 'ABCD1234',
  title: 'Testing',
  body: 'Body testing'
};
const initState = {
  auth: {},
  ui: {
    loading: false,
    msgError: null
  },
  notes: {
    notes: {},
    active: note
  }
};

const store = mockStore(initState);
store.dispatch = jest.fn();

describe('Tests NoteScreen component', () => {

  const wrapper = mount(
    <Provider store={store}>
      <NoteScreen />
    </Provider>
  );

  test('Should display correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should call activeNote', () => {
    expect(activeNote).toHaveBeenCalled();
    expect(activeNote).toHaveBeenCalledTimes(1);
    expect(activeNote).toHaveBeenCalledWith(note.id, note);
  });

  test('Should call activeNote with input change event', () => {
    const title = 'Test title';

    act(() => {
      wrapper.find('input[name="title"]').prop('onChange')({
        target: {
          value: title,
          name: 'title'
        }
      });
    });

    expect(activeNote).toHaveBeenCalled();
    expect(activeNote).toHaveBeenCalledTimes(2);
    expect(activeNote).toHaveBeenLastCalledWith(note.id, { ...note, title });
  });

  test('Should call startDeleting', () => {
    wrapper.find('button.btn.btn-danger').prop('onClick')();

    expect(startDeleting).toHaveBeenCalled();
    expect(startDeleting).toHaveBeenCalledTimes(1);
  });
});

