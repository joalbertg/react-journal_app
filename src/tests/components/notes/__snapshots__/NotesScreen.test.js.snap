// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests NoteScreen component Should display correctly 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      6th Monday 2020
    </span>
    <input
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value="Testing"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happened today"
      value="Body testing"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
